src/cli/remove.rs

```diff
@@ -44,40 +57,67 @@ pub async fn execute(args: Args) -> miette::Result<()> {
+                    Ok(pkg_name) => {
+                        if let Some(p) = &args.platform {
+                            DependencyRemovalResult::PyPi(
+                                project.manifest.remove_target_pypi_dependency(&pkg_name, p),
+                            )
+                        } else {
+                            DependencyRemovalResult::PyPi(
+                                project.manifest.remove_pypi_dependency(&pkg_name),
+                            )
+                        }
+                    }
+                    Err(e) => DependencyRemovalResult::Error(e.into()),
+                }
             } else {
-                project.manifest.remove_dependency(dep, &spec_type)
+                DependencyRemovalResult::Conda(if let Some(p) = &args.platform {
+                    project
+                        .manifest
+                        .remove_target_dependency(dep, &spec_type, p)
+                } else {
+                    project.manifest.remove_dependency(dep, &spec_type)
+                })
             }
         })
-        .collect::<Vec<_>>();
+        .collect::<Vec<DependencyRemovalResult>>();
 
     project.save()?;
 
     // updating prefix after removing from toml
     let _ = get_up_to_date_prefix(&project, LockFileUsage::Update).await?;
-
-    for (removed, spec) in results.iter().flatten() {
-        let table_name = if let Some(p) = &args.platform {
-            format!("target.{}.{}", p.as_str(), spec_type.name())
-        } else {
-            spec_type.name().to_string()
-        };
-
+    let table_name = if let Some(p) = &args.platform {
+        format!("target.{}.{}", p.as_str(), spec_type.name())
+    } else {
+        spec_type.name().to_string()
+    };
+    fn print_ok_dep_removal(pkg_name: &String, pkg_extras: &String, table_name: &String) {
         eprintln!(
             "Removed {} from [{}]",
-            console::style(format!("{removed} {spec}")).bold(),
-            console::style(table_name).bold(),
-        );
+            console::style(format!("{pkg_name} {pkg_extras}")).bold(),
+            console::style(table_name).bold()
+        )
     }
-
-    for result in &results {
-        if let Err(e) = result {
-            eprintln!("{e}");
+    for result in results.iter() {
+        match result {
+            DependencyRemovalResult::Conda(Ok(pixi_result)) => print_ok_dep_removal(
+                &pixi_result.0.to_string(),
+                &pixi_result.1.to_string(),
+                &table_name,
+            ),
+            DependencyRemovalResult::PyPi(Ok(pypi_result)) => print_ok_dep_removal(
+                &pypi_result.0.as_str().to_string(),
+                &pypi_result.1.to_string(),
+                &table_name,
+            ),
+            DependencyRemovalResult::Conda(Err(e))
+            | DependencyRemovalResult::PyPi(Err(e))
+            | DependencyRemovalResult::Error(e) => {
+                return Err(miette!("{e}"));
```

Comment:
The problem with this is that it will early out, before printing all errors and results. Normally this is indeed what you would do but as the rest of the code assumes running over all packages the user gives I would not return in this for loop.

Reply:
fixed